modules/user/user.model.js

```diff
@@ -65,4 +72,19 @@ const userSchema = new mongoose.Schema({
   confirmationToken: String,
 });
 
+userSchema.pre('findOneAndDelete', async function(next) {
+  const id = this.getQuery()['_id'];
+  const userRole = await this.model.findById(id);
+
+  if (!userRole) {
```

Comment:
I don't think that we need this particular check because it is a big chance that before calling the "delete" we perform some check in the business logic and it should be performed there, not here. Here we just need to ignore cases where we cannot find a user

Reply:
fixed